testing/testcafe/tests/editors/calendar/calendar.ts

```diff
@@ -0,0 +1,83 @@
+import { createScreenshotsComparer } from 'devextreme-screenshot-comparer';
+import url from '../../../helpers/getPageUrl';
+import createWidget from '../../../helpers/createWidget';
+import { changeTheme } from '../../../helpers/changeTheme';
+
+fixture`Calendar`
+  .page(url(__dirname, '../../container.html'))
+  .afterEach(async () => {
+    await changeTheme('generic.light');
+  });
+
+const themes = ['generic.light', 'generic.light.compact', 'material.blue.light', 'material.blue.light.compact'];
+themes.forEach((theme) => {
+  test(`Calendar with showWeekNumbers rendered correct (${theme})`, async (t) => {
+    const { takeScreenshot, compareResults } = createScreenshotsComparer(t);
+
+    await t
+      .expect(await takeScreenshot(`calendar-with-showWeekNumbers-theme=${theme.replace(/\./g, '-')}.png`, '#container'))
+      .ok()
+      .expect(compareResults.isValid())
+      .ok(compareResults.errorMessages());
+  }).before(async () => {
+    await changeTheme(theme);
+
+    await createWidget('dxCalendar', {
+      value: new Date('01.01.2022'),
+      showWeekNumbers: true,
+    });
+  });
+});
+
+test('Calendar with showWeekNumbers rendered correct for last week of year value', async (t) => {
+  const { takeScreenshot, compareResults } = createScreenshotsComparer(t);
+
+  await t
+    .expect(await takeScreenshot('calendar-with-showWeekNumbers-last-week.png', '#container'))
+    .ok()
+    .expect(compareResults.isValid())
+    .ok(compareResults.errorMessages());
+}).before(async () => {
+  await createWidget('dxCalendar', {
```

Comment:
Why the `last week` number is `1` but not `54` in your scenario. I think that logic to calculate week number works incorrectly, we need double check it.

Why? 
In Google Calendar for december of 2021 year I see that the last week number is 52.
<img width="240" alt="image" src="https://user-images.githubusercontent.com/15689080/191172201-7b56da46-0dee-474c-a8ea-26ce6e93bcd7.png">

Reply:
And you variant 
<img width="298" alt="image" src="https://user-images.githubusercontent.com/15689080/191172287-e62bedf7-20ef-4333-82ad-ada9e092ff50.png">

Reply:
it's because of american system of counting week numbers

Reply:
Ok, let's create a howTo ticket to demonstrate the approved solution how we can change the system using cellTemplate